sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -122,10 +126,20 @@ class HadoopTableReader(
     val attrsWithIndex = attributes.zipWithIndex
     val mutableRow = new SpecificInternalRow(attributes.map(_.dataType))
 
-    val deserializedHadoopRDD = hadoopRDD.mapPartitions { iter =>
+    val deserializedHadoopRDD = hadoopRDD.mapPartitionsWithIndex { (index, iter) =>
       val hconf = broadcastedHadoopConf.value.value
       val deserializer = deserializerClass.newInstance()
       deserializer.initialize(hconf, tableDesc.getProperties)
+      if (skipHeaderLineCount > 0 && isTextInputFormatTable) {
+        val partition = hadoopRDD.partitions(index).asInstanceOf[HadoopPartition]
+        if (partition.inputSplit.t.asInstanceOf[FileSplit].getStart() == 0) {
```

Comment:
is `partition.inputSplit.t.asInstanceOf[FileSplit].getStart() != 0` tested (for a split that does not start from the start of the file, we do not skip)?